src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,517 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, anim, 0.0f, 0.0f, 0.0f, 2, 0.0f); // Anim Mode == STOP
+    EnKarebaba_ResetCollider(this);
+    this->actor.shape.rot.x = -0x4000;
+    this->actor.params = 200;
+    this->actor.parent = NULL;
+    this->actor.shape.unk_10 = 0.0f;
+    Math_Vec3f_Copy(&this->actor.posRot.pos, &this->actor.initPosRot.pos);
+    this->actionFunc = EnKarebaba_Dead;
+}
+
+void EnKarebaba_SetupRegrow(EnKarebaba* this) {
+    this->actor.shape.unk_08 = 0.0f;
+    this->actor.shape.unk_10 = 22.0f;
+    this->headCollider.dim.radius = sHeadCollider.dim.radius;
+    Actor_SetScale(&this->actor, 0.0f);
+    this->actionFunc = EnKarebaba_Regrow;
+}
+
+void EnKarebaba_Grow(EnKarebaba* this, GlobalContext* globalCtx) {
+    f32 scale;
+
+    this->actor.params++;
+    scale = this->actor.params * 0.05f;
+    Actor_SetScale(&this->actor, 0.005f * scale);
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + (14.0f * scale);
+    if (this->actor.params == 20) {
+        EnKarebaba_SetupIdle(this);
+    }
+}
+
+void EnKarebaba_Idle(EnKarebaba* this, GlobalContext* globalCtx) {
+    if (this->actor.xzDistFromLink < 200.0f && fabsf(this->actor.yDistFromLink) < 30.0f) {
+        EnKarebaba_SetupAwaken(this);
+    }
+}
+
+void EnKarebaba_Awaken(EnKarebaba* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_ApproxF(&this->actor.scale.x, 0.01f, 0.0005f);
+    this->actor.scale.y = this->actor.scale.z = this->actor.scale.x;
+    if (Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y + 60.0f, 5.0f)) {
+        EnKarebaba_SetupUpright(this);
+    }
+    this->actor.shape.rot.y += 0x1999;
+    EffectSsHahen_SpawnBurst(globalCtx, &this->actor.initPosRot.pos, 3.0f, 0, 12, 5, 1, -1, 10, 0);
+}
+
+void EnKarebaba_Upright(EnKarebaba* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    if (func_800A56C8(&this->skelAnime, 0.0f) || func_800A56C8(&this->skelAnime, 12.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKU_JR_MOUTH);
+    }
+
+    // Check if Link attacks the Deku Baba
+    if (this->bodyCollider.base.acFlags & 2) {
+        EnKarebaba_SetupDying(this);
+        func_80032C7C(globalCtx, &this->actor);
+    } else if (Math_Vec3f_DistXZ(&this->actor.initPosRot.pos, &player->actor.posRot.pos) > 240.0f) {
+        EnKarebaba_SetupRetract(this);
+    } else if (this->actor.params == 0) {
+        EnKarebaba_SetupSpin(this);
+    }
+}
+
+void EnKarebaba_Spin(EnKarebaba* this, GlobalContext* globalCtx) {
+    s32 value;
+    f32 thing;
+
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (func_800A56C8(&this->skelAnime, 0.0f) || func_800A56C8(&this->skelAnime, 12.0f)) {
+        if (true) { } // Here for matching purposes only.
```

Comment:
```suggestion
        if (1) {} // Here for matching purposes only.
```
very small thing but that's how all `if (1)`s are formatted